domain/robstoll-lib/extensions/kotlin_1_3/atrium-domain-robstoll-lib-kotlin_1_3-common/src/main/kotlin/ch/tutteli/atrium/domain.robstoll.lib.kotlin_1_3/creating/resultAssertions.kt

```diff
@@ -1,5 +1,15 @@
 package ch.tutteli.atrium.domain.robstoll.lib.kotlin_1_3.creating
 
-fun _isSuccess() {
-    TODO("fix signature and implement, only here so that module.info does not complain")
-}
+import ch.tutteli.atrium.creating.Expect
+import ch.tutteli.atrium.domain.builders.ExpectImpl
+import ch.tutteli.atrium.domain.creating.changers.ExtractedFeaturePostStep
+import ch.tutteli.atrium.translations.DescriptionResultAssertion
+
+fun  <E, T : Result<E>>_isSuccess( assertionContainer: Expect<T>) : ExtractedFeaturePostStep<T, E> =
+    ExpectImpl.feature.extractor(assertionContainer)
+        .withDescription(DescriptionResultAssertion.VALUE)
+        .withRepresentationForFailure(DescriptionResultAssertion.ILLEGAL_STATE_EXCEPTION)
+        .withCheck { it.isSuccess }
+        .withFeatureExtraction { it.getOrNull() ?: throw IllegalStateException("checked it.isSuccess and suddenly it isn't any more: $it")}
```

Comment:
I think we need to use a different strategy, `E` could be a nullable type and thus getOrNull could return `null` even if it isSuccess. I suggest you add such a case to the spec to verify. Once you verified that my assumption is correct, then we have to use an unchecked cast instead `it.getOrNull() as E`

Reply:
I have change the API of FeatureExtractor, please rebase on master and remove `.withCheck { it.isSuccess }`
and instead use
```
.withFeatureExtraction {
    if (it.isSuccess) {
        @Suppress(/* fine as we just checked that it is success */ "UNCHECKED_CAST")
        Some(it as E)
    } else {
        None
    }
}
```